### DISCUSSION TITLE
Cannot connect to cloud instance inside a Kubernetes pod

### DISCUSSION THREAD

**Discussion by ItsRobbAllen** [ravendb/ravendb#14429] [Q&A] [answered]
I am completely stuck. 

I have microservices that need to connect to Raven. Locally, they connect to a docker image running on my dev machine. Or, if I point them to a cloud instance, they run just fine as well.

When I deploy it out to Kubernetes however, the Raven client bombs out with the following error

`Unhandled exception. System.AggregateException: Failed to retrieve database topology from all known nodes.
https://a.xxx-dev.xxx.ravendb.cloud -> The request for https://a.xxx-dev.xxx.ravendb.cloud/topology?name=xxx.Master&first-topology-update&applicationIdentifier=6570b4a2-03d7-4da7-821f-71e4ed534737 failed with timeout after 00:00:15 -> A task was canceled. (The request for https://a.xxx-dev.xxx.ravendb.cloud/topology?name=xxx.Master&first-topology-update&applicationIdentifier=6570b4a2-03d7-4da7-821f-71e4ed534737 failed with timeout after 00:00:15)
 ---> System.TimeoutException: The request for https://a.xxx-dev.xxx.ravendb.cloud/topology?name=xxx.Master&first-topology-update&applicationIdentifier=6570b4a2-03d7-4da7-821f-71e4ed534737 failed with timeout after 00:00:15
 ---> System.Threading.Tasks.TaskCanceledException: A task was canceled.`

Is there something I need to do in Kubernetes to allow a pod to contact Raven Cloud? Is there a special Egress setting? Does the client need Ingress to allow the Raven server to connect back to the pod?

It's not a DNS issue because I can connect to other things on the Internet like Google, and it doesn't work even if I use the IP address of the server directly. I just can't figure out why my service is blind to Raven Cloud.

**Comment by ayende** [comment]
The first thing that comes to mind is if you have your cloud setup limit IP addresses? 
That is the usual reason for those sort of issues.

If not that, check the DNS settings and see where you are trying to connect _to_.

**Reply by ItsRobbAllen** [reply]
I installed

`kubectl apply -f https://k8s.io/examples/admin/dns/dnsutils.yaml`

and running tests gives me
```
$ kubectl exec -i -t dnsutils -- nslookup a.xxx-dev.xxx.ravendb.cloud
Server:         10.0.0.10
Address:        10.0.0.10#53

Non-authoritative answer:
Name:   a.xxx-dev.xxx.ravendb.cloud
Address: 55.164.135.123
```

So DNS seems fine at the cluster level. I'm going to create a tiny pod to test with to see if it's just *that* particular pod or something else. If anyone has any other things to look at, I'm all ears.

**Comment by ItsRobbAllen** [comment, marked as answer]
Figured it out!

Someone had changed the _outbound_ IP on the load balancer and that IP was not in the allowed list.

That was it.
